packages/aws-cdk/bin/cdk.ts

```diff
@@ -169,19 +164,27 @@ async function initCommandLine() {
       print(`Toolkit stack: ${colors.bold(toolkitStackName)}`);
     }
 
+    const provisioner = new CloudFormationDeploymentTarget({ aws, toolkitStackName });
+
     args.STACKS = args.STACKS || [];
     args.ENVIRONMENTS = args.ENVIRONMENTS || [];
+    const stackNames = args.STACKS;
 
-    const cli = new CdkToolkit({ appStacks, provisioner });
+    async function cli() {
+      const stacks = await appStacks.selectStacks(
+        stackNames,
+        args.exclusively ? ExtendedStackSelection.None : ExtendedStackSelection.Upstream);
+      return new CDKToolkit({ stacks, provisioner });
```

Comment:
I find it weird that all commands must go through this CDKToolkit class which is in cdk-deploy...

Reply:
Yes, it is a weird setup currently - i did the minimum work to separate them logically. The next step is to wrap `cdk-deploy` in a CLI interface and call it from `aws-cdk` instead of routing through this hacky shared dependency.

Did you have thoughts on where the `diff` command should go? It seems like it has similar security requirements as `deploy`, so I left it in `CDKToolkit` class as part of `cdk-deploy`.

Reply:
Makes sense. I like it that you are taking the smallest step possible.

I think "diff" needs to go in the main CLI, as well as CDKToolkit. It only needs READ permissions from the account (similar to env context providers), while deploy needs ADMIN WRITE permissions (that's the main motivation for separation).